tests/put_get.go

```diff
@@ -0,0 +1,221 @@
+// is consistent accross nodes.
+type PutGetTest struct {
+	clients             []*minio.Client
+	logFile             *os.File
+	bucket              string
+	objectsWithMetaData sync.Map
+	apiStats            *Stats
+	concurrency         int
+}
+
+// metaData to store the uploaded object's info
+// which will be used for verification.
+type metaData struct {
+	sha256 []byte
+	size   int64
+	etag   string
+}
+
+func (t *PutGetTest) bucketName() string {
+	return t.bucket
+}
+
+// Name - name of the test.
+func (t *PutGetTest) Name() string {
+	return "PutGet"
+}
+
+// Init initialized the test.
+func (t *PutGetTest) Init(ctx context.Context, config Config, stats *Stats) error {
+	t.clients = config.Clients
+	t.logFile = config.LogFile
+	t.bucket = config.Bucket
+	t.objectsWithMetaData = sync.Map{}
+	t.apiStats = stats
+	t.concurrency = config.Concurrency
+	return nil
+}
+
+// Setup prepares the test by uploading the objects to the test bucket.
+func (t *PutGetTest) Setup(ctx context.Context) error {
+	return runFn(ctx,
+		objectCountForPutGetTest,
+		t.concurrency,
+		func(ctx context.Context, index int) error {
+			client, err := utils.GetRandomClient(t.clients)
+			if err != nil {
+				return err
+			}
+			object := fmt.Sprintf("confess/%s/%s/pfx-%d", t.Name(), time.Now().Format("01_02_06_15:04"), index)
+			hasher := sha256.New()
+			tr := io.TeeReader(reader(humanize.MiByte*5), hasher)
+			if objectInfo, err := put(ctx, putConfig{
+				client:     client,
+				bucketName: t.bucket,
+				objectName: object,
+				reader:     tr,
+				opts:       minio.PutObjectOptions{},
+			}, t.apiStats); err != nil {
+				if err := log(ctx,
+					t.logFile,
+					t.Name(),
+					client.EndpointURL().String(),
+					fmt.Sprintf("unable to put the object %s; %v", object, err),
+				); err != nil {
+					return err
+				}
+			} else {
+				t.objectsWithMetaData.Store(object, metaData{
+					sha256: hasher.Sum(nil),
+					size:   objectInfo.Size,
+					etag:   objectInfo.ETag,
+				})
+			}
+			return nil
+		},
+	)
+}
+
+// Run executes the test by verifying if the get output is
+// consistent accross the nodes.
+func (t *PutGetTest) Run(ctx context.Context) error {
+	var offlineNodes atomic.Int64
+	var errFound bool
+	if err := runFn(ctx,
+		len(t.clients),
+		t.concurrency,
+		func(ctx context.Context, index int) error {
+			if t.clients[index].IsOffline() {
+				offlineNodes.Add(1)
+				if err := log(
+					ctx,
+					t.logFile,
+					t.Name(),
+					t.clients[index].EndpointURL().String(),
+					fmt.Sprintf("the node %s is offline", t.clients[index].EndpointURL().String()),
+				); err != nil {
+					return err
+				}
+				return nil
+			}
+			if err := t.verify(ctx, t.clients[index]); err != nil {
+				if err := log(
+					ctx,
+					t.logFile,
+					t.Name(),
+					t.clients[index].EndpointURL().String(),
+					fmt.Sprintf("unable to verify %s test; %v", t.Name(), err),
+				); err != nil {
+					return err
+				}
+				errFound = true
```

Comment:
here you will need to check if verify failed because a node is currently offline and avoid marking as errFound. The tool should be resilient against node down etc and only report actual consistency errors

Reply:
Only online nodes will be picked for verification @poornas 

If the node's result is not matching, we will log that err to the logfile and continue verifying other active nodes. This boolean is to know if the overall test run succeeded or not and to respond back with the appropriate error.

Reply:
verify also makes a get call - and node can go offline during that step. As long as you are not marking those failures as errors, it will be fine

Reply:
understood, will fix